token-erc-20/chaincode-java/src/main/java/org/example/TokenERC20Contract.java

```diff
@@ -0,0 +1,519 @@
+
+        String currentBalanceBytes = stub.getStringState(balanceKey.toString().trim());
+        if (currentBalanceBytes.isBlank() || currentBalanceBytes.length() == 0) {
+            throw new ChaincodeException("The balance does not exist");
+        }
+        long currentBalance = Long.valueOf(currentBalanceBytes.toString());
+        long updatedBalance = currentBalance - amountInt;
+
+        stub.putStringState(balanceKey.toString().trim(), String.valueOf(updatedBalance));
+
+        // Decrease totalSupply
+        String totalSupplyBytes = stub.getStringState(totalSupplyKey.toString().trim());
+        if (totalSupplyBytes.isBlank() || totalSupplyBytes.length() == 0) {
+            throw new ChaincodeException("totalSupply does not exist.");
+        }
+        long totalSupply = Long.parseLong(totalSupplyBytes.toString()) - amountInt;
+        stub.putStringState(totalSupplyKey.toString().trim(), String.valueOf(totalSupply));
+
+        // Emit the Transfer event
+
+        JSONObject obj = new JSONObject();
+        obj.put("from", minter);
+        obj.put("to", "0x0");
+        obj.put("value", amountInt);
+        stub.setEvent("Transfer", this.serialize(obj));
+        System.out.printf("minter account %s balance updated from %d to %d", minter, currentBalance, updatedBalance);
+        return true;
+    }
+
+    /**
+     * ClientAccountBalance returns the balance of the requesting client's account.
+     *
+     * @param {Context} ctx the transaction context
+     * @returns {Number} Returns the account balance
+     */
+
+    @Transaction()
+    public long getClientAccountBalance(final Context ctx) {
+        // Get ID of submitting client identity
+        ChaincodeStub stub = ctx.getStub();
+        String clientAccountID = ctx.getClientIdentity().getId();
+        CompositeKey balanceKey = stub.createCompositeKey(balancePrefix, clientAccountID);
+        String balanceBytes = stub.getStringState(balanceKey.toString().trim());
+        if (balanceBytes.isBlank() || balanceBytes.length() == 0) {
+
+            String errorMessage = String.format("the account  %s does not exist", clientAccountID);
+            throw new ChaincodeException(errorMessage);
+        }
+        long balance = Long.parseLong(balanceBytes.trim());
+
+        return balance;
+    }
+
+    // ClientAccountID returns the id of the requesting client's account.
+    // In this implementation, the client account ID is the clientId itself.
+    // Users can use this function to get their own account id, which they can then
+    // give to others as the payment address
+    @Transaction()
+    public String getClientAccountID(final Context ctx) {
+        // Get ID of submitting client identity
+        String clientAccountID = ctx.getClientIdentity().getId();
+        return clientAccountID;
+    }
+
+    private byte[] serialize(Object object) {
+        String jsonStr = new JSONObject(object).toString();
+        return jsonStr.getBytes(UTF_8);
```

Comment:
This is only ever called with a `JSONObject` as the argument.
```suggestion
    private byte[] serialize(final JSONObject object) {
        return object.toString().getBytes(UTF_8);
```